// @elgato-stream-deck/core/dist/id.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MODEL_NAMES = exports.DeviceModelId = void 0;
var DeviceModelId;
(function (DeviceModelId) {
    DeviceModelId["ORIGINAL"] = "original";
    DeviceModelId["ORIGINALV2"] = "originalv2";
    DeviceModelId["ORIGINALMK2"] = "original-mk2";
    DeviceModelId["ORIGINALMK2SCISSOR"] = "original-mk2-scissor";
    DeviceModelId["MINI"] = "mini";
    DeviceModelId["XL"] = "xl";
    DeviceModelId["PEDAL"] = "pedal";
    DeviceModelId["PLUS"] = "plus";
    DeviceModelId["NEO"] = "neo";
    DeviceModelId["STUDIO"] = "studio";
    DeviceModelId["MODULE6"] = "6-module";
    DeviceModelId["MODULE15"] = "15-module";
    DeviceModelId["MODULE32"] = "32-module";
    DeviceModelId["NETWORK_DOCK"] = "network-dock";
    DeviceModelId["GALLEON_K100"] = "galleon-k100";
    DeviceModelId["PLUS_XL"] = "plus-xl";
})(DeviceModelId || (exports.DeviceModelId = DeviceModelId = {}));
exports.MODEL_NAMES = {
    [DeviceModelId.ORIGINAL]: 'Stream Deck',
    [DeviceModelId.MINI]: 'Stream Deck Mini',
    [DeviceModelId.XL]: 'Stream Deck XL',
    [DeviceModelId.ORIGINALV2]: 'Stream Deck',
    [DeviceModelId.ORIGINALMK2]: 'Stream Deck MK.2',
    [DeviceModelId.ORIGINALMK2SCISSOR]: 'Stream Deck MK.2 (Scissor)',
    [DeviceModelId.PLUS]: 'Stream Deck +',
    [DeviceModelId.PEDAL]: 'Stream Deck Pedal',
    [DeviceModelId.NEO]: 'Stream Deck Neo',
    [DeviceModelId.STUDIO]: 'Stream Deck Studio',
    [DeviceModelId.MODULE6]: 'Stream Deck 6 Module',
    [DeviceModelId.MODULE15]: 'Stream Deck 15 Module',
    [DeviceModelId.MODULE32]: 'Stream Deck 32 Module',
    [DeviceModelId.NETWORK_DOCK]: 'Stream Deck Network Dock',
    [DeviceModelId.GALLEON_K100]: 'Galleon K100 SD',
    [DeviceModelId.PLUS_XL]: 'Stream Deck + XL',
};
