// @elgato-stream-deck/core/dist/models/15-key.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamDeck15KeyFactory = StreamDeck15KeyFactory;
const base_js_1 = require("./base.js");
const generic_gen2_js_1 = require("./generic-gen2.js");
const id_js_1 = require("../id.js");
const controlsGenerator_js_1 = require("../controlsGenerator.js");
const base15KeyProperties = {
    SUPPORTS_RGB_KEY_FILL: true,
    CONTROLS: (0, controlsGenerator_js_1.freezeDefinitions)((0, controlsGenerator_js_1.generateButtonsGrid)(5, 3, { width: 72, height: 72 })),
    KEY_SPACING_HORIZONTAL: 25,
    KEY_SPACING_VERTICAL: 25,
    FULLSCREEN_PANELS: 1,
    HAS_NFC_READER: false,
    SUPPORTS_CHILD_DEVICES: false,
};
function StreamDeck15KeyFactory(model, device, options, _tcpPropertiesService) {
    const properties = {
        ...base15KeyProperties,
        MODEL: model,
        PRODUCT_NAME: id_js_1.MODEL_NAMES[model],
    };
    const services = (0, generic_gen2_js_1.createBaseGen2Properties)(device, options, properties, null);
    return new base_js_1.StreamDeckBase(device, options, services);
}
