// @elgato-stream-deck/core/dist/models/32-key.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamDeck32KeyFactory = StreamDeck32KeyFactory;
const base_js_1 = require("./base.js");
const generic_gen2_js_1 = require("./generic-gen2.js");
const id_js_1 = require("../id.js");
const controlsGenerator_js_1 = require("../controlsGenerator.js");
const base32KeyProperties = {
    SUPPORTS_RGB_KEY_FILL: true,
    CONTROLS: (0, controlsGenerator_js_1.freezeDefinitions)((0, controlsGenerator_js_1.generateButtonsGrid)(8, 4, { width: 96, height: 96 })),
    KEY_SPACING_HORIZONTAL: 32,
    KEY_SPACING_VERTICAL: 39,
    FULLSCREEN_PANELS: 1,
    HAS_NFC_READER: false,
    SUPPORTS_CHILD_DEVICES: false,
};
function StreamDeck32KeyFactory(model, device, options, _tcpPropertiesService) {
    const properties = {
        ...base32KeyProperties,
        MODEL: model,
        PRODUCT_NAME: id_js_1.MODEL_NAMES[model],
    };
    const services = (0, generic_gen2_js_1.createBaseGen2Properties)(device, options, properties, null);
    return new base_js_1.StreamDeckBase(device, options, services);
}
