// @elgato-stream-deck/core/dist/models/6-key.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamDeck6KeyFactory = StreamDeck6KeyFactory;
const generic_gen1_js_1 = require("./generic-gen1.js");
const id_js_1 = require("../id.js");
const controlsGenerator_js_1 = require("../controlsGenerator.js");
const imageWriter_js_1 = require("../services/imageWriter/imageWriter.js");
const headerGenerator_js_1 = require("../services/imageWriter/headerGenerator.js");
const base6KeyProperties = {
    SUPPORTS_RGB_KEY_FILL: false, // TODO - verify this
    CONTROLS: (0, controlsGenerator_js_1.freezeDefinitions)((0, controlsGenerator_js_1.generateButtonsGrid)(3, 2, { width: 80, height: 80 })),
    KEY_SPACING_HORIZONTAL: 28,
    KEY_SPACING_VERTICAL: 28,
    FULLSCREEN_PANELS: 0,
};
function StreamDeck6KeyFactory(model, device, options, _tcpPropertiesService) {
    const properties = {
        ...base6KeyProperties,
        MODEL: model,
        PRODUCT_NAME: id_js_1.MODEL_NAMES[model],
    };
    return (0, generic_gen1_js_1.StreamDeckGen1Factory)(device, options, properties, new imageWriter_js_1.StreamdeckDefaultImageWriter(new headerGenerator_js_1.StreamdeckGen1ImageHeaderGenerator()), { colorMode: 'bgr', rotate: true, yFlip: true }, 2835);
}
