// @elgato-stream-deck/core/dist/models/generic-gen1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamDeckGen1Factory = StreamDeckGen1Factory;
const base_js_1 = require("./base.js");
const gen1_js_1 = require("../services/properties/gen1.js");
const default_js_1 = require("../services/buttonsLcdDisplay/default.js");
const bitmap_js_1 = require("../services/imagePacker/bitmap.js");
const callback_hook_js_1 = require("../services/callback-hook.js");
const gen1_js_2 = require("../services/input/gen1.js");
function extendDevicePropertiesForGen1(rawProps) {
    return {
        ...rawProps,
        KEY_DATA_OFFSET: 0,
        HAS_NFC_READER: false,
        SUPPORTS_CHILD_DEVICES: false,
    };
}
function StreamDeckGen1Factory(device, options, properties, imageWriter, targetOptions, bmpImagePPM) {
    const fullProperties = extendDevicePropertiesForGen1(properties);
    const events = new callback_hook_js_1.CallbackHook();
    return new base_js_1.StreamDeckBase(device, options, {
        deviceProperties: fullProperties,
        events,
        properties: new gen1_js_1.Gen1PropertiesService(device),
        buttonsLcd: new default_js_1.DefaultButtonsLcdService(imageWriter, new bitmap_js_1.BitmapButtonLcdImagePacker(targetOptions, bmpImagePPM), device, fullProperties),
        lcdSegmentDisplay: null,
        inputService: new gen1_js_2.ButtonOnlyInputService(fullProperties, events),
        encoderLed: null,
    });
}
