// @elgato-stream-deck/core/dist/models/original.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamDeckOriginalFactory = StreamDeckOriginalFactory;
const generic_gen1_js_1 = require("./generic-gen1.js");
const id_js_1 = require("../id.js");
const imageWriter_js_1 = require("../services/imageWriter/imageWriter.js");
const controlsGenerator_js_1 = require("../controlsGenerator.js");
const originalProperties = {
    MODEL: id_js_1.DeviceModelId.ORIGINAL,
    PRODUCT_NAME: id_js_1.MODEL_NAMES[id_js_1.DeviceModelId.ORIGINAL],
    SUPPORTS_RGB_KEY_FILL: false,
    CONTROLS: (0, controlsGenerator_js_1.freezeDefinitions)((0, controlsGenerator_js_1.generateButtonsGrid)(5, 3, { width: 72, height: 72 }, true)),
    KEY_SPACING_HORIZONTAL: 25,
    KEY_SPACING_VERTICAL: 25,
    FULLSCREEN_PANELS: 0,
};
function StreamDeckOriginalFactory(device, options) {
    return (0, generic_gen1_js_1.StreamDeckGen1Factory)(device, options, originalProperties, new imageWriter_js_1.StreamdeckOriginalImageWriter(), { colorMode: 'bgr', xFlip: true }, 3780);
}
